File: mobile-apps/ring_fit_record/frontend/lib/models/record.dart
class Record {
  final String attachment;
  final String comfort;
  final String memo;
  final DateTime createdAt;

  Record({
    required this.attachment,
    required this.comfort,
    required this.memo,
    required this.createdAt,
  });

  factory Record.fromJson(Map<String, dynamic> json) {
    DateTime createdAt = DateTime.parse(json['CreatedAt']);
    return Record(
      attachment: json['Attachment'],
      comfort: json['Comfort'],
      memo: json['Memo'],
      createdAt: createdAt,
    );
  }
}

File: mobile-apps/ring_fit_record/backend/internal/server/funcs/record.go
package funcs

import (
	"backend/internal/db"
	dbmodels "backend/internal/db/models"
	"backend/internal/server/models"
	"net/http"
	"time"

	"github.com/gin-gonic/gin"
)

func GetRecordsHandler(ctx *gin.Context) {
	records := db.GetRecords()
	ctx.JSON(http.StatusOK, gin.H{
		"records": records,
	})
}

func RegisterRecordHandler(ctx *gin.Context) {
	var req models.RegisterRecord
	if err := ctx.ShouldBindJSON(&req); err != nil {
		ctx.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
		return
	}
	err := db.RegisterRecord(dbmodels.Record{Attachment: req.Attachment, Comfort: req.Comfort, Memo: req.Memo, CreatedAt: time.Now()})
	if err != nil {
		ctx.JSON(http.StatusInternalServerError, gin.H{
			"error": err,
		})
	} else {
		ctx.JSON(http.StatusOK, gin.H{})
	}
}

func DeleteRecordHandler(ctx *gin.Context) {
	docID := ctx.Param("id")
	if err := db.DeleteRecord(docID); err != nil {
		ctx.JSON(http.StatusInternalServerError, gin.H{"error": err})
		return
	}

	ctx.JSON(http.StatusOK, gin.H{})
}
